src/corefx/System.Globalization.Native/pal_collation.c

```diff
@@ -184,19 +157,18 @@ static UCharList* GetCustomRules(int32_t options, UColAttributeValue strength, i
     }
 
     // If we need to create customRules, the KanaType custom rule will be 88 kana characters * 4 = 352 chars long
-    // and the Width custom rule will be at least 215 halfwidth characters * 4 = 860 chars long.
+    // and the Width custom rule will be at most 212 halfwidth characters * 5 = 1060 chars long.
```

Comment:
>212 [](start = 49, length = 3)

How did you get 212 here? 

looking at [https://unicode.org/charts/PDF/UFF00.pdf](https://unicode.org/charts/PDF/UFF00.pdf), If considering all half width characters then this number would be really 201. 

Also, the Kana characters should be 96?

I am ok to use a bigger numbers as I am not expecting the allocation will be much bigger anyway.

Reply:
linking the original PR introduced creating this custom rule [https://github.com/dotnet/coreclr/pull/2196](https://github.com/dotnet/coreclr/pull/2196)

---
In reply to: [277037987](https://github.com/dotnet/coreclr/pull/24100#discussion_r277037987) [](ancestors = 277037987)

Reply:
> How did you get 212 here?

The constant `g_HalfFullCharsLength` declared in the file. The description could be wrong though.

> Also, the Kana characters should be 96?

Taken from the actual loop and comments in the same function.